docker/README.md

```diff
@@ -0,0 +1,175 @@
+# Puppet-bolt container
+
+Puppet bolt it available as a docker image. This document covers some possible ways to use the container.
+
+## Completely stand-alone
+
+When running Bolt from the container, the `localhost` target is the container environment (not the docker host environment). The following example shows running a command against the localhost target in the container.
+```
+cas@cas-ThinkPad-T460p:~/working_dir/docker_bolt$ docker run puppet-bolt command run 'cat /etc/os-release' -t localhost
+Started on localhost...
+Finished on localhost:
+  STDOUT:
+    NAME="Ubuntu"
+    VERSION="16.04.6 LTS (Xenial Xerus)"
+    ID=ubuntu
+    ID_LIKE=debian
+    PRETTY_NAME="Ubuntu 16.04.6 LTS"
+    VERSION_ID="16.04"
+    HOME_URL="http://www.ubuntu.com/"
+    SUPPORT_URL="http://help.ubuntu.com/"
+    BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
+    VERSION_CODENAME=xenial
+    UBUNTU_CODENAME=xenial
+Successful on 1 node: localhost
+Ran on 1 node in 0.00 seconds
+```
+In order to pass connection information and custom module content we need to share data from the host with the container. The following sections describe some ways to accomplish sharing data with the puppet bolt container.
+
+## Pass Inventory as an environment variable
+
+In the case where no custom module content is required for the bolt action you wish to execute with the container and the only information you need is how to connect to targets you can pass inventory as an enviornment variable. The following inventory has all the information needed to connect to the example target. 
+
+```yaml
+---
+nodes:
+  - name: pnz2rzpxfzp95hh.delivery.puppetlabs.net
+    alias: docker-example
+    config:
+      transport: ssh
+      ssh:
+        user: root
+        password: secret-password
+        host-key-check: false
+```
+The following invocation shows an example of running the built-in `facts` task against the target listed in inventory. 
+```
+cas@cas-ThinkPad-T460p:~/working_dir/docker_bolt$ docker run --env "BOLT_INVENTORY=$(cat Boltdir/inventory.yaml)" puppet-bolt task run facts -t docker-example
```

Comment:
I had to do `docker run --network host -e "BOLT_INVENTORY=$(cat ./inventory.yaml)" puppet-bolt task run facts -t docker-example` - might be worth adding?